### PR TITLE
fix compile error

### PR DESCRIPTION
**Pull request by liruipeng**
This PR fixes compile errors/warnings on lassen.

- [x] tux
- [x] lassen
- [x] tioga

### CODE REVIEW

**Review by victorapm — APPROVED** [review]
Thanks for the fixes!

File: src/parcsr_ls/par_ge_device.c [hunk 1]
```diff
@@ -105,9 +105,9 @@ hypre_GaussElimSetupDevice(hypre_ParAMGData *amg_data,
 
 #else
    /* Silence declared but never referenced warnings */
-   (A_piv  = NULL);
-   (A_mat  = NULL);
-   (A_work = NULL);
```
Comment (waynemitchell) [inline src/parcsr_ls/par_ge_device.c:110]: What was the issue with setting these to `NULL`? Just curious. I find it funny that we keep modifying this code that is really just here to get rid of some compiler warnings.

**Review by waynemitchell — APPROVED** [review]

File: src/parcsr_ls/par_ge_device.c (hunk 1 above)
Comment (liruipeng) [reply src/parcsr_ls/par_ge_device.c:110]: The previous code doesn’t work for the compilers used on lassen (still having the set-but-not-used warning). In general, I don't like to trick the compilers in a way to silence warnings. Warnings are there for reasons. I would rather write "the most accurate and robust" code, even if it's longer. This "fix" seems to work fine and a minimum change to what we have now. I would also like to hear suggestions from @rfalgout @ulrikeyang @oseikuffuor1 .  Thanks.

Comment (waynemitchell) [reply src/parcsr_ls/par_ge_device.c:110]: Oh, weird. I am the one who put in the `= NULL` version of this, and I thought I had gotten a clean lassen regression with this change in. But this must have been a last minute change that I didn't test... thanks for fixing!

Comment (rfalgout) [reply src/parcsr_ls/par_ge_device.c:110]: This seems reasonable to me.  The alternative would be to put the variables in an #if as well.  I doubt that's cleaner.

Comment (liruipeng) [reply src/parcsr_ls/par_ge_device.c:110]: > Oh, weird. I am the one who put in the `= NULL` version of this, and I thought I had gotten a clean lassen regression with this change in. But this must have been a last minute change that I didn't test... thanks for fixing!

Strange... This was the warning:
```
/usr/tce/packages/cuda/cuda-10.1.243/bin/nvcc -ccbin=mpixlC -gencode arch=compute_70,code=sm_70  -g -O0 -lineinfo -expt-extended-lambda -std=c++11 --x cu -Xcompiler "-g "  -DHAVE_CONFIG_H -I.. -I../distributed_ls/Euclid -I. -I./.. -I./../blas -I./../lapack -I./../multivector -I./../utilities -I./../krylov -I./../seq_mv -I./../parcsr_mv -I./../distributed_matrix -I./../matrix_matrix -I./../IJ_mv -I./../parcsr_block_mv   -I/usr/tce/packages/cuda/cuda-10.1.243/include         -c par_ge_device.c -o par_ge_device.obj
par_ge_device.c(32): warning: variable "A_piv" was set but never used

par_ge_device.c(33): warning: variable "A_mat" was set but never used

par_ge_device.c(34): warning: variable "A_work" was set but never used

par_ge_device.c(226): warning: variable "A_mat" was set but never used

par_ge_device.c(227): warning: variable "A_piv" was set but never used
```
Do we use the same compiler, flags, etc?

Comment (waynemitchell) [reply src/parcsr_ls/par_ge_device.c:110]: > Do we use the same compiler, flags, etc?

I just ran the `machine-lassen.sh` script to test, I didn't do any of my own configure/compile.

**Pull request merged**